### ISSUE TITLE
Values loaded by NCDatasets backend with "_Unsigned" CF attribute don't correspond to Xarray

### ISSUE THREAD

**Issue by asinghvi17** [rafaqz/Rasters.jl#735]
MWE:
```julia
using FilePathsBase, AWSS3
download(p"s3://noaa-goes16/ABI-L2-SSTF/2020/210/00/OR_ABI-L2-SSTF-M6_G16_s20202100000205_e20202100059513_c20202100105456.nc", "noaa.nc")
using Rasters
r1 = Raster("noaa.nc"; lazy = true)
heatmap(r1)
```
![iTerm2 ZHWLfL](https://github.com/user-attachments/assets/6d03b04d-6a11-4a9a-80b4-3e6dc85217b5)
```julia
heatmap(read(r1))
```
![iTerm2 LvJSGK](https://github.com/user-attachments/assets/17f7f5cd-542a-4886-b241-17b97219c067)

None of this agrees with what xarray loads:
![download](https://github.com/user-attachments/assets/8476362c-2407-4d47-8706-850ca25bca85)

but it looks like all CF conventions are applied...

**Comment by rafaqz** [comment]
Those just look like different colour schemes?

The first one should call `read` in the Rasters recipe? Does it even hit that?

**Comment by asinghvi17** [comment]
No the color scheme is the same, the color limits are different though and it seems the numbers are different as well.

**Comment by asinghvi17** [comment]
Plus you can see the structure is totally strange before `read`

**Comment by rafaqz** [comment]
Yes read is clearly a problem. The dispatch is wrong somewhere it looks like broken iteration in DiskArrays.

But the non lazy plots without noise are just NCDatasets.jl vs XArray, Rasters is not affecting the values

**Comment by asinghvi17** [comment]
aha ok, will check out what's going on there.  This is on the cf branch so hopefully it's not ignoring some obscure CF convention...

**Comment by rafaqz** [comment]
Makes way more sense this is on `cf`. You gave me a fright that that mess is from main...

**Comment by rafaqz** [comment]
Both or just one

**Comment by rafaqz** [comment]
The DiskArrays munging I will look at tonight, seems bad if that's on main

**Comment by asinghvi17** [comment]
You don't for this, it's public access.

**Comment by felixcremer** [comment]
I get a AWS Credentials not found error when I try your reproducer. 
```julia
julia> download(p"s3://noaa-goes16/ABI-L2-SSTF/2020/210/00/OR_ABI-L2-SSTF-M6_G16_s20202100000205_e20202100059513_c20202100105456.nc", "noaa.nc")
ERROR: Can't find AWS credentials!

Stacktrace:
  [1] AWS.AWSCredentials(; profile::Nothing, throw_cred_error::Bool)
    @ AWS ~/.julia/packages/AWS/SqWtW/src/AWSCredentials.jl:145
  [2] AWSCredentials
    @ ~/.julia/packages/AWS/SqWtW/src/AWSCredentials.jl:113 [inlined]
  [3] AWS.AWSConfig()
    @ AWS ~/.julia/packages/AWS/SqWtW/src/AWSConfig.jl:32
  [4] #global_aws_config#91
    @ ~/.julia/packages/AWS/SqWtW/src/AWS.jl:81 [inlined]
  [5] global_aws_config
    @ ~/.julia/packages/AWS/SqWtW/src/AWS.jl:79 [inlined]
  [6] get_config
    @ ~/.julia/packages/AWSS3/8cxdr/src/s3path.jl:143 [inlined]
  [7] exists(fp::S3Path{Nothing})
    @ AWSS3 ~/.julia/packages/AWSS3/8cxdr/src/s3path.jl:244
  [8] cp(src::S3Path{Nothing}, dst::PosixPath; force::Bool, follow_symlinks::Bool)
    @ FilePathsBase ~/.julia/packages/FilePathsBase/Laclt/src/path.jl:560
  [9] cp
    @ ~/.julia/packages/FilePathsBase/Laclt/src/path.jl:551 [inlined]
 [10] download(url::S3Path{Nothing}, localfile::PosixPath)
    @ FilePathsBase ~/.julia/packages/FilePathsBase/Laclt/src/path.jl:697
 [11] download(url::S3Path{Nothing}, localfile::String)
    @ FilePathsBase ~/.julia/packages/FilePathsBase/Laclt/src/path.jl:700
 [12] top-level scope
    @ REPL[5]:1
```

**Comment by asinghvi17** [comment]
Which OS/DiskArrays versions are you on?

**Comment by felixcremer** [comment]
I am on linux and DiskArrays 0.3.23

**Comment by felixcremer** [comment]
How did you manage to get DiskArrays 0.4 are you on the PR branch from Fabian? 
Can you plot the differences between xarray and Raster?

**Comment by asinghvi17** [comment]
Couldn't get the wrapped arrays to work together because of a dimension mismatch (`Dim{:x}` v/s `X`) but `collect` saved the day:

![download-7](https://github.com/user-attachments/assets/1aebfbbe-744e-420c-9983-171cd9f7be97)

**Comment by asinghvi17** [comment]
There's clearly some structure here that I bet is described by some CF attribute.  But I don't know what it is!

**Comment by rafaqz** [comment]
Haha we have ever issue with every problem in branch of DiskArrays/CommonDataModel showing up here!

Can we keep issues here to registered versions of all deps? I will go insane otherwise

**Comment by asinghvi17** [comment]
Does Rasters check the `valid_range` attribute at all?

**Comment by rafaqz** [comment]
On main Rasters checks nothing, in CF I don't think it checks valid_range

**Comment by asinghvi17** [comment]
It turns out there's an attribute `_Unsigned` in CF which controls whether the type is signed or unsigned?!!  No idea why they don't just encode it in the type like normal people

Turns out that was the issue.  Might be worth a CommonDataModel.jl issue...

**Comment by rafaqz** [comment]
Absolutely worth a CommonDataModel.jl big report

**Comment by asinghvi17** [comment]
The DiskArrays portion of this issue was fixed by https://github.com/meggart/DiskArrays.jl/pull/198.   The unsigned attribute likely needs to be handled in NCDatasets.jl

**Comment by rafaqz** [comment]
This now depends on https://github.com/JuliaGeo/CommonDataModel.jl/issues/24
